### DISCUSSION TITLE
footnote support

### DISCUSSION THREAD

**Discussion by gureckis** [vuejs/vitepress#704] [Q&A] [answered]
Hi,
Awesome project and the new theme in 1.0.0-alpha is slick.  I was wondering if anyone know if it is possible to enable markdown footnotes in vitepress?  I tried searching the past issues, discussion, and code and couldn't find any reference to it.  In longer docs can be really helpful!
Thanks,
Todd

**Comment by brc-dd** [comment, marked as answer]
Please refer advanced configuration section here: https://vitepress.vuejs.org/guide/markdown.html#advanced-configuration

After installing and configuring `markdown-it-footnote` like this:

```js
// .vitepress/config.js

import footnote from 'markdown-it-footnote'
import { defineConfig } from 'vitepress'

export default defineConfig({
  markdown: {
    config: (md) => {
      md.use(footnote)
    }
  }
})
```

You'll be able to use footnotes:

```
Here's a simple footnote,[^1] and here's a longer one.[^bignote]

[^1]: This is the first footnote.

[^bignote]: Here's one with multiple paragraphs and code.

    Indent paragraphs to include them in the footnote.

    `{ my code }`

    Add as many paragraphs as you like.
```

Output:

![image](https://user-images.githubusercontent.com/40380293/171682575-fa9f6172-1ead-4b09-837e-3dcf4663aa93.png)

**Reply by gureckis** [reply]
awesome, worked perfectly!  Thanks so much!

**Comment by arcqiufeng** [comment]
I know little about js.

So can you tell me how to merge the code:

I have the code already:

```js
    markdown: {
        config: (md) => {
            md.use(mathjax3); // 添加公式latex支持
        },
    },
```

how to merge the code into?

```js
markdown: {
    config: (md) => {
      md.use(footnote)
    }
  }
```

**Comment by arcqiufeng** [comment]
However, I got it.

Just add the line
```
markdown: {
    config: (md) => {
        md.use(mathjax3);      
        md.use(footnote)
    }
  }
```

**Comment by khawarizmus** [comment]
@brc-dd is there any way to specify the location of the footnotes? it always renders at the bottom of the page

**Reply by brc-dd** [reply]
Ah, that's what "foot"note means 🤷‍♂️ You'd need to find some other plugin probably if you just wanna extract them and render manually.
